server/etcdserver/apply/uber_applier_test.go

```diff
@@ -0,0 +1,244 @@
+			request:     &pb.InternalRaftRequest{Range: &pb.RangeRequest{}},
+			expectError: errors.ErrCorrupt,
+		},
+		{
+			name:        "DeleteRange request returns ErrCorrupt after alarm CORRUPT is activated",
+			request:     &pb.InternalRaftRequest{DeleteRange: &pb.DeleteRangeRequest{}},
+			expectError: errors.ErrCorrupt,
+		},
+		{
+			name:        "Txn request returns ErrCorrupt after alarm CORRUPT is activated",
+			request:     &pb.InternalRaftRequest{Txn: &pb.TxnRequest{}},
+			expectError: errors.ErrCorrupt,
+		},
+		{
+			name:        "Compaction request returns ErrCorrupt after alarm CORRUPT is activated",
+			request:     &pb.InternalRaftRequest{Compaction: &pb.CompactionRequest{}},
+			expectError: errors.ErrCorrupt,
+		},
+		{
+			name:        "LeaseGrant request returns ErrCorrupt after alarm CORRUPT is activated",
+			request:     &pb.InternalRaftRequest{LeaseGrant: &pb.LeaseGrantRequest{}},
+			expectError: errors.ErrCorrupt,
+		},
+		{
+			name:        "LeaseRevoke request returns ErrCorrupt after alarm CORRUPT is activated",
+			request:     &pb.InternalRaftRequest{LeaseRevoke: &pb.LeaseRevokeRequest{}},
+			expectError: errors.ErrCorrupt,
+		},
+	}
+
+	ua := defaultUberApplier(t)
+	result := ua.Apply(&pb.InternalRaftRequest{
+		Header: &pb.RequestHeader{},
+		Alarm: &pb.AlarmRequest{
+			Action:   pb.AlarmRequest_ACTIVATE,
+			MemberID: memberId,
+			Alarm:    pb.AlarmType_CORRUPT,
+		},
+	}, true)
+	require.NotNil(t, result)
+	require.Nil(t, result.Err)
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			result = ua.Apply(tc.request, true)
+			require.NotNil(t, result)
+			require.Equalf(t, tc.expectError, result.Err, "Apply: got %v, expect: %v", result.Err, tc.expectError)
+		})
+	}
+}
+
+// TestUberApplier_Alarm_Quota tests the applier returns ErrNoSpace after alarm NOSPACE is activated
+func TestUberApplier_Alarm_Quota(t *testing.T) {
+	tcs := []struct {
+		name        string
+		request     *pb.InternalRaftRequest
+		expectError error
+	}{
+		{
+			name:        "Put request returns ErrCorrupt after alarm NOSPACE is activated",
+			request:     &pb.InternalRaftRequest{Put: &pb.PutRequest{Key: []byte(key)}},
+			expectError: errors.ErrNoSpace,
+		},
+		{
+			name: "Txn request cost > 0 returns ErrCorrupt after alarm NOSPACE is activated",
+			request: &pb.InternalRaftRequest{Txn: &pb.TxnRequest{
+				Success: []*pb.RequestOp{
+					{
+						Request: &pb.RequestOp_RequestPut{
+							RequestPut: &pb.PutRequest{
+								Key: []byte(key),
+							},
+						},
+					},
+				}}},
+			expectError: errors.ErrNoSpace,
+		},
+		{
+			name: "Txn request cost = 0 is still allowed after alarm NOSPACE is activated",
+			request: &pb.InternalRaftRequest{Txn: &pb.TxnRequest{
+				Success: []*pb.RequestOp{
+					{
+						Request: &pb.RequestOp_RequestRange{
+							RequestRange: &pb.RangeRequest{
+								Key: []byte(key),
+							},
+						},
+					},
+				}}},
+			expectError: nil,
+		},
```

Comment:
Could you please add an conditional delete in the Txn test case? It was the motivation of PR https://github.com/etcd-io/etcd/pull/13435/

Reply:
Sure. Lemme check the background first.

Reply:
I can add a test case, but tbh I don't think that's the part of function I am trying to cover here......a better place would be the unit test of the [server/storage](https://github.com/etcd-io/etcd/tree/833aabe4cd5061c02cb72d010836c7e57510aab2/server/storage) package. However that package has no test coverage either. So nevermind, I will add it.